extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/metrics/HibernateMpMetricsProvider.java

```diff
@@ -0,0 +1,185 @@
+package io.quarkus.hibernate.orm.runtime.metrics;
+
+import java.util.function.ToLongFunction;
+
+import javax.annotation.Priority;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.metrics.*;
+import org.eclipse.microprofile.metrics.annotation.RegistryType;
+import org.hibernate.SessionFactory;
+import org.hibernate.stat.Statistics;
+
+import io.quarkus.hibernate.orm.runtime.JPAConfig;
+import io.quarkus.runtime.StartupEvent;
+
+/**
+ * This is only discovered if
+ */
+public class HibernateMpMetricsProvider {
+    private static final String SESSION_FACTORY_TAG_NAME = "entityManagerFactory";
```

Comment:
`persistence-unit` would sound more suitable to me for the tag key

Reply:
Meh. I used what micrometer uses. Allows metrics to be aggregated across spring/micronaut/quarkus runtimes.

Reply:
Ah, now I get it, ok then